src/views/SystemIntake/index.tsx

```diff
@@ -98,6 +99,12 @@ export const SystemIntake = ({ match }: SystemIntakeProps) => {
     return null;
   };
 
+  const dispatchSave = () => {
+    if (id) {
```

Comment:
I wonder if instead of using the ref (because of object equality, if it could be more reliable to use the following since Formik is doing the value comparison.

```suggestion
    if (id && formikRef.current.dirty) {
```

ref: https://jaredpalmer.com/formik/docs/api/formik#dirty-boolean